modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -249,34 +249,38 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
   def cmdStateO(qid: QueueId): Optional[EngineState, BatchCommandState] =
     queueO(qid) ^|-> ExecutionQueue.cmdState
 
-  private def addSeqs(qid: QueueId, seqIds: List[Observation.Id]): Endo[EngineState] = st => (
-    for {
-      q    <- st.queues.get(qid)
-      seqs <- seqIds.filter(sid => st.executionState.sequences.get(sid).map(seq => !seq.status.isRunning && !seq.status.isCompleted && !q.queue.contains(sid)).getOrElse(false)).some.filter(_.nonEmpty)
-      if (q.status(st) =!= BatchExecState.Running)
-    } yield queueO(qid).modify(_.addSeqs(seqs))(st)
-  ).getOrElse(st)
-
-  def addSequencesToQueue(q: EventQueue, qid: QueueId, seqIds: List[Observation.Id]): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
-    Event.modifyState[executeEngine.ConcreteTypes]((addSeqs(qid, seqIds) withEvent UpdateQueueAdd(qid, seqIds)).toHandle)
+  private def addSeqs(qid: QueueId, seqIds: List[Observation.Id]): executeEngine.HandleType[Unit] =
+    executeEngine.get.flatMap{ st => (
+      for {
+        q <- st.queues.get(qid)
+        seqs <- seqIds.filter(sid => st.executionState.sequences.get(sid)
+          .map(seq => !seq.status.isRunning && !seq.status.isCompleted && !q.queue.contains(sid))
+          .getOrElse(false)).some.filter(_.nonEmpty)
+        if (!seqs.isEmpty)
+      } yield executeEngine.modify(queueO(qid).modify(_.addSeqs(seqs))) *>
+        ((q.cmdState, q.status(st)) match {
+        case (_, BatchExecState.Completed)       => ((EngineState.queues ^|-? index(qid) ^|-> ExecutionQueue.cmdState)
+          .set(BatchCommandState.Idle) >>> {(_, ())}).toHandle
```

Comment:
So this changes the cmdState right?

Reply:
Yes, back to `Idle`, which makes the `execState` change to `Idle`.